src/libstd/lib.rs

```diff
@@ -249,30 +249,48 @@ pub mod num;
 /* Runtime and platform support */
 
 #[macro_use]
-pub mod thread_local;
+pub mod thread;
+
+#[stable(feature = "rust1", since = "1.0.0")]
+#[deprecated(since = "1.0.0", reason = "use std::thread::{LocalKey, ScopedKey} instead")]
+/// Deprecated: use `std::thread::{LocalKey, ScopedKey}` instead.
+pub mod thread_local {
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[deprecated(since = "1.0.0", reason = "use std::thread::LocalKey instead")]
+    /// Deprecated: use `std::threaed::LocalKey` instead.
+    pub use thread::LocalKey as Key;
+
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[deprecated(since = "1.0.0", reason = "use std::thread::LocalKeyState instead")]
+    /// Deprecated: use `std::threaed::LocalKeyState` instead.
+    pub use thread::LocalKeyState as State;
+
+    #[unstable(feature = "std_misc")]
+    #[deprecated(since = "1.0.0", reason = "use std::thread::ScopedKey instead")]
+    /// Deprecated: use `std::threaed::ScopedKey` instead.
+    pub mod scoped {
+        #[unstable(feature = "std_misc")]
+        #[deprecated(since = "1.0.0", reason = "use std::thread::ScopedKey instead")]
+        /// Deprecated: use `std::threaed::ScopedKey` instead.
+        pub use thread::ScopedKey as Key;
```

Comment:
Due to `pub use` not actually issuing deprecation warnings, and the fact that this is such a relatively rarely imported module, I'd be fine just deleting this module.